iceoryx_posh/doc/chunk_header.md

```diff
@@ -0,0 +1,218 @@
+payloadAddress = align(headerEndAddress, payloadAlignment);
+payloadOffset = payloadAddress - addressof(chunkHeader);
+```
+
+3. Custom header is used
+
+```
+headerEndAddress = addressof(chunkHeader) + sizeof(chunkHeader) + sizeof(customHeader);
+potentialBackOffsetAddress = align(headerEndAddress, alignof(payloadOffset));
+potentialPayloadAddress = potentialBackOffsetAddress + sizeof(payloadOffset);
+payloadAddress = align(potentialPayloadAddress, payloadAlignment);
+payloadOffset = payloadAddress - addressof(chunkHeader);
+```
+
+### Required Chunk Size Calculation
+
+In order to fit the custom header and the payload into he chunk, a worst case calculation has to be done. We can assume that a chunk is aligned to 32 bytes, which is also the alignment of the `ChunkHeader`.
+
+1. No custom header and payload alignment doesn't exceed the `ChunkHeader` alignment
+
+```
+chunkSize = sizeof(chunkHeader) + payloadSize;
+```
+
+2. No custom header and payload alignment exceeds the `ChunkHeader` alignment
+
+Worst case scenario is when a part of the `ChunkHeader` crosses the payload alignment boundary, so that the payload must be aligned to the next boundary.
+```
+sizeLeftOfThePayloandAlignmentBoundary = sizeof(chunkHeader) - alignof(chunkHeader);
+chunkSize = sizeLeftOfThePayloandAlignmentBoundary + payloadAlignment + payloadSize;
+```
+
+3. Custom header is used
+
+Similar to case 2, but in this case it is the `back-offset` which might cross the payload alignment boundary.
+```
+headerSize = sizeof(chunkHeader) + sizeof(customHeader)
+sizeLeftOfThePayloandAlignmentBoundary = align(headerSize, alignof(payloadOffset));
+maxAlignment = max(alignof(payloadOffset), payloadAlignment);
+chunkSize = sizeLeftOfThePayloandAlignmentBoundary + maxAlignment + payloadSize;
+```
+
+### Accessing Chunk Header Extension
+
+The `ChunkHeader` has a template method to get custom header. There is a risk to use this wrong by accident, but there is currently no better solution for this.
+
+Since the custom header is always adjacent to the `ChunkHeader`, the formula to get the custom header is:
+```
+customHeader = addressOf(chunkHeader) + sizeof(chunkHeader);
+```
+
+### ChunkHeader Methods
+
+- `void* payload()` returns a pointer to the payload
+- `template <typename T> T* customHeader()` returns a pointer to the custom header
+- `static ChunkHeader* fromPayload(const void* const payload)` returns a pointer to the `ChunkHeader` associated to the payload
+
+### Integration Into Publisher/Subscriber API
+
+The `Publisher` has additional template parameters for the custom header and payload alignment. Alternatively this could also be done with the `allocate` method, but that increases the risk of using it wrong by accident.
+
+Additionally, to fully integrate the custom header into the regular control flow, a pointer to a `ChunkHeaderHook` can be passed to the `Publisher` c'tor.
+
+The `ChunkHeaderHook` is not in the shared memory and has the following virtual methods:
```

Comment:
Can you write a rough example how the publisher and subscriber class and c'tor signatures would look like

Reply:
sure. I will put that in the `Open Issues` section, since this might change when it's actually used in an example and we notice that it feels weird to use.

Reply:
done